front50-core/src/main/java/com/netflix/spinnaker/front50/echo/Notification.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.front50.echo;
+
+import java.util.Collection;
+import java.util.Map;
+import lombok.Data;
+import lombok.NonNull;
+
+@Data
+public class Notification {
+  @NonNull Type notificationType;
+  @NonNull Collection<String> to;
+  @NonNull String templateGroup;
+  @NonNull Severity severity;
+  @NonNull Source source;
+  @NonNull Map<String, ?> additionalContext;
+
+  @Data
+  public static class Source {
+    @NonNull String application;
+  }
+
+  public enum Type {
+    EMAIL
+  }
```

Comment:
Wondering aloud: Do we want this to be an enum? If Echo creates a new notification type, it means that we'll need to also update this enum as well.

Reply:
In `echo`, this is actually typed as a `String` (the enum was removed in March by Clay because of plugins).  I don't feel strongly that this should be an enum... it would make as much sense for it to be a constant value.  I'll update.

Reply:
Actually, on second thought, I think I do prefer this as an enum.  I think it conveys the types of notifications that Front50 will send, not the set of types that Echo supports.  Changing to a constant felt more awkward, and putting it inline when creating the notification felt more error-prone.